app/models/category.rb

```diff
@@ -0,0 +1,5 @@
+class Category < ApplicationRecord
+  has_many  :sub_categories, dependent: :restrict_with_error
```

Comment:
test its message for user

Reply:
yes shows sms for user and do not delete